quickshop-bukkit/src/main/java/com/ghostchu/quickshop/util/ShopUtil.java

```diff
@@ -102,20 +99,32 @@ public static void setPrice(final QuickShop plugin, @NotNull QUser user, final d
     }
 
     final PriceLimiterCheckResult checkResult = limiter.check(user, shop.getItem(), plugin.getCurrency(), price);
+    final String currency = shop.getCurrency() == null ? (plugin.getCurrency() == null ? "" : plugin.getCurrency()) : shop.getCurrency();
 
     switch (checkResult.getStatus()) {
       case PRICE_RESTRICTED -> {
-        plugin.text().of(user.getUniqueId(), "restricted-prices", Util.getItemStackName(shop.getItem()),
-                Component.text(checkResult.getMin()),
-                Component.text(checkResult.getMax())).send();
+        final double min = checkResult.getMin();
+        final double max = checkResult.getMax();
+
+        if (min > 0 && max >= 0) {
+          plugin.text().of(user.getUniqueId(), "restricted-prices", Util.getItemStackName(shop.getItem()),
+                  currency + min,
+                  currency + max).send();
+        } else if (min > 0) {
+          plugin.text().of(user.getUniqueId(), "restricted-price-min", Util.getItemStackName(shop.getItem()),
+                  currency + min).send();
+        } else {
+          plugin.text().of(user.getUniqueId(), "restricted-price-max", Util.getItemStackName(shop.getItem()),
+                  currency + max).send();
+        }
         return;
       }
       case REACHED_PRICE_MIN_LIMIT -> {
-        plugin.text().of(user, "price-too-cheap", (format) ? MsgUtil.decimalFormat(checkResult.getMin()) : Double.toString(checkResult.getMin())).send();
+        plugin.text().of(user, "price-too-cheap", currency + checkResult.getMin()).send();
```

Comment:
This is bad, use economy formatter. Different economic plug-ins have different requirements for formatting currencies, and directly adding them up will break the currency formatting